File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string,vector<string>>m;
        for(auto str: strs)
        {
            string st = str;
            sort(st.begin(), st.end());
            m[st].push_back(str);
        }
        vector<vector<string>>res;
        for(auto i : m)
        {
            res.push_back(i.second);
        }
        return res;
    }
};

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        d = defaultdict(list)
        for s in strs:
            temp = "".join(sorted(s))
            d[temp].append(s)
        return [d[r] for r in d]